assigning a local var ptr to an interface var fails

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
	"net"
)

func main() {
	addr := net.TCPAddr{IP: net.IPv4(1,1,1,1), Port: 80}
	var s fmt.Stringer = &addr
	fmt.Println(s.String())
}
```

Expected result:
```console
$ go run ./sample.go
1.1.1.1:80
```

Got:
```console
$ yaegi ./sample.go
run.go:601: ./sample.go:10:23: genInterfaceWrapper error, no method String
./sample.go:9:10: panic
runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000332000, 0x40)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00022fe10)
	/Users/berng/go/src/github.com/containous/yaegi/interp/interp.go:320 +0xbc
panic(0x18deee0, 0x214c1b0)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc00030a0a0, 0xc000217200)
	/Users/berng/go/src/github.com/containous/yaegi/interp/run.go:113 +0x20c
panic(0x18deee0, 0x214c1b0)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/stdlib._fmt_Stringer.String(0x0, 0x0, 0x0)
	/Users/berng/go/src/github.com/containous/yaegi/stdlib/go1_14_fmt.go:111 +0x22
reflect.Value.call(0x18eaf20, 0xc000206890, 0x293, 0x19bc6e7, 0x4, 0x2196618, 0x0, 0x0, 0xc00022fb70, 0x1051b2c, ...)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x18eaf20, 0xc000206890, 0x293, 0x2196618, 0x0, 0x0, 0xc00022fc10, 0x1263cea, 0x18eaf20)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:321 +0xb4
github.com/containous/yaegi/interp.callBin.func7(0xc00030a0a0, 0xc00020e600)
	/Users/berng/go/src/github.com/containous/yaegi/interp/run.go:958 +0x1a2
github.com/containous/yaegi/interp.runCfg(0xc000217200, 0xc00030a0a0)
	/Users/berng/go/src/github.com/containous/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00021c180, 0xc000216900, 0xc00030a000)
	/Users/berng/go/src/github.com/containous/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00021c180, 0xc000312000, 0xa3, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/berng/go/src/github.com/containous/yaegi/interp/interp.go:388 +0x41f
main.main()
	/Users/berng/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:94 +0x548```

--- a/_test/bin5.go
+++ b/_test/bin5.go
@@ -1,0 +1,15 @@
+package main
+
+import (
+	"fmt"
+	"net"
+)
+
+func main() {
+	addr := net.TCPAddr{IP: net.IPv4(1, 1, 1, 1), Port: 80}
+	var s fmt.Stringer = &addr
+	fmt.Println(s.String())
+}
+
+// Output:
+// 1.1.1.1:80

--- a/_test/fun14.go
+++ b/_test/fun14.go
@@ -9,5 +9,5 @@ func main() {
 	println(b, i)
 }
 
-// Output
+// Output:
 // true 2

--- a/_test/new2.go
+++ b/_test/new2.go
@@ -8,3 +8,6 @@ func main() {
 	a := f()
 	println(*(a.(*int)))
 }
+
+// Output:
+// 0

--- a/_test/struct37.go
+++ b/_test/struct37.go
@@ -15,3 +15,6 @@ func main() {
 	_, err := c.Get("url")
 	println(strings.Contains(err.Error(), "unsupported protocol scheme"))
 }
+
+// Output:
+// true

--- a/_test/struct45.go
+++ b/_test/struct45.go
@@ -14,4 +14,5 @@ func main() {
 	println(t.b)
 }
 
-// Output: true
+// Output:
+// true

--- a/_test/type22.go
+++ b/_test/type22.go
@@ -13,3 +13,6 @@ type T struct {
 func main() {
 	println(foo().Name)
 }
+
+// Output:
+// foo

--- a/interp/run.go
+++ b/interp/run.go
@@ -594,9 +594,13 @@ func genInterfaceWrapper(n *node, typ reflect.Type) func(*frame) reflect.Value {
 		w := reflect.New(wrap).Elem()
 		for i, m := range methods {
 			if m == nil {
+				if r := v.MethodByName(names[i]); r.IsValid() {
+					w.Field(i).Set(r)
+					continue
+				}
 				o := vv.FieldByIndex(indexes[i])
 				if r := o.MethodByName(names[i]); r.IsValid() {
-					w.Field(i).Set(o.MethodByName(names[i]))
+					w.Field(i).Set(r)
 				} else {
 					log.Println(n.cfgErrorf("genInterfaceWrapper error, no method %s", names[i]))
 				}

